packages/pods-core/src/builtin/NotionPod.ts

```diff
@@ -0,0 +1,156 @@
+      properties: {
+        apiKey: {
+          type: "string",
+          description: "Api key for Notion",
+        },
+        vault: {
+          type: "string",
+          description: "vault to export from",
+        },
+      },
+    }) as JSONSchemaType<NotionExportConfig>;
+  }
+
+  /**
+   * Method to create pages in Notion
+   */
+  createPagesInNotion = async (blockPagesArray: any, notion: Client) => {
+    // rate limiter method
+    const sendRequest = async () => {
+      blockPagesArray.forEach(async (block: any) => {
+        // @ts-ignore
+        await limiter.removeTokens(1);
+        try {
+          await notion.pages.create(block);
+        } catch (error) {
+          this.L.error({
+            msg: "failed to export all the notes.",
+            payload: error,
+          });
+          throw new DendronError({
+            message: JSON.stringify(error),
+            severity: ERROR_SEVERITY.MINOR,
+          });
+        }
+      });
+    };
+    sendRequest();
+  };
+
+  /**
+   * Method to convert markdown to Notion Block
+   */
+  convertMdToNotionBlock = (notes: NoteProps[], pageId: string) => {
+    const notionBlock = notes.map((note) => {
+      const children = markdownToBlocks(note.body);
+      return {
+        parent: {
+          page_id: pageId,
+        },
+        properties: {
+          title: {
+            title: [{ type: "text", text: { content: note.title } }],
+          },
+        },
+        children,
+      };
+    });
+    return notionBlock;
+  };
+
+  /**
+   * Method to get page name of a Notion Page
+   */
+  getPageName = (page: Page) => {
+    const { title } =
+      page.parent.type !== "database_id"
+        ? (page.properties.title as TitlePropertyValue)
+        : (page.properties.Name as TitlePropertyValue);
+    return title[0] ? title[0].plain_text : "Untitled";
+  };
+
+  /**
+   * Method to get all the pages from Notion
+   */
+  getAllNotionPages = async (notion: Client) => {
+    const allDocs = await notion.search({
+      sort: { direction: "descending", timestamp: "last_edited_time" },
+      filter: { value: "page", property: "object" },
+    });
+    const pagesMap: any = {};
+    const pages = allDocs.results as Page[];
+    pages.forEach((page: Page) => {
+      const key = this.getPageName(page);
+      const value = page.id;
+      pagesMap[key] = value;
+    });
+    return pagesMap;
+  };
+
+  async plant(opts: ExportPodPlantOpts) {
+    const { config, utilityMethods } = opts;
+    const { getSelectionFromQuickpick, withProgressOpts } =
+      utilityMethods as NotionUtilMethods;
+    const { apiKey, vault } = config as NotionExportConfig;
+    let { notes } = opts;
+
+    notes = notes.filter((note) => note.vault.fsPath === vault);
+    // Initializing a client
+    const notion = new Client({
+      auth: apiKey,
+    });
+
+    const pagesMap = await withProgressOpts.withProgress(
+      {
+        location: withProgressOpts.location,
```

Comment:
nit: not necessary, but it'd be cool if we pass the progress object into getAllNotionPages.  And then on pages.forEach, we can update the progress every 10% or something like that. 

Feel free to push this off for later.